doc/design/operator-bundle.md

```diff
@@ -0,0 +1,144 @@
+
+This example uses [Operator Registry Manifests](https://github.com/operator-framework/operator-registry#manifest-format) format to build an operator bundle image. The source directory of an operator registry bundle has the following layout.
+```
+$ tree test
+test
+├── 0.1.0
+│   ├── testbackup.crd.yaml
+│   ├── testcluster.crd.yaml
+│   ├── testoperator.v0.1.0.clusterserviceversion.yaml
+│   └── testrestore.crd.yaml
+└── annotations.yaml
+```
+
+### Bundle Dockerfile
+
+This is an example of a `Dockerfile` for operator bundle:
+```
+FROM scratch
+
+# We are pushing an operator-registry bundle
+# that has both metadata and manifests.
+LABEL operators.operatorframework.io.bundle.resources=manifests+metadata
+LABEL operators.operatorframework.io.bundle.mediatype=registry+v1
+
+ADD test/0.1.0 /manifests
+ADD test/annotations.yaml /metadata/annotations.yaml
+```
+
+Below is the directory layout of the operator bundle inside the image:
+```bash
+$ tree
+/
+├── manifests
+│   ├── testbackup.crd.yaml
+│   ├── testcluster.crd.yaml
+│   ├── testoperator.v0.1.0.clusterserviceversion.yaml
+│   └── testrestore.crd.yaml
+└── metadata
+    └── annotations.yaml
+```
+
+## Operator Bundle Command-Line Tool
+
+A CLI tool is available to generate Bundle annotations and Dockerfile based on provided operator manifests.
+
+### Bundle CLI
+
+In order to build `operator-cli` CLI tool, follow these steps:
+
+1. Clone [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) repository.
+2. Build `operator-cli` binary:
+```bash
+$ go build ./cmd/operator-cli/
+```
+
+Now, a binary named `operator-cli` is available in OLM's directory to use.
+```bash
+$ ./operator-cli
+Generate operator bundle metadata and build bundle image.
+
+Usage:
+   bundle [command]
+
+Available Commands:
+  build       Build operator bundle image
+  generate    Generate operator bundle metadata and Dockerfile
+
+Flags:
+  -h, --help   help for bundle
+
+Use " bundle [command] --help" for more information about a command.
+```
+
+### Generate Bundle Annotations and DockerFile
+
+Using `operator-cli` CLI, bundle annotations can be generated from provided operator manifests. The command for `generate` task is:
+```bash
+$ ./operator-cli bundle generate --directory /test/0.0.1/
+```
+The `--directory` or `-d` specifies the directory where the operator manifests are located. The `annotations.yaml` and `Dockerfile` are generated in the same directory where the manifests folder is located (not where the YAML manifests are located). For example:
+```bash
+$ tree test
+test
+├── 0.0.1
+│   ├── testbackup.crd.yaml
+│   ├── testcluster.crd.yaml
+│   ├── testoperator.v0.1.0.clusterserviceversion.yaml
+│   └── testrestore.crd.yaml
+├── annotations.yaml
+└── Dockerfile
+```
+
+Note: If there are `annotations.yaml` and `Dockerfile` existing in the directory, they will be overwritten.
```

Comment:
Won't these files also be overwritten with the build command given [this line](https://github.com/operator-framework/operator-lifecycle-manager/pull/1053/files#diff-3218ce33478ac109a3ae6300ab2b9d54R91)?